source/common/http/async_client_impl.h

```diff
@@ -360,6 +364,9 @@ class AsyncRequestImpl final : public AsyncClient::Request,
   void onReset() override;
 
   // Http::StreamDecoderFilterCallbacks
+  void addDecodedData(Buffer::Instance&, bool) override {
```

Comment:
Also I think I'm confused - where is this called?  I thought the addDecodedData you added down in the router was on the HCM not the async client.  Mind clueing me in on what I'm missing?

Reply:
The HTTP async client uses the router filter internally, so the async client implements all of the callback interfaces and effectively pretends to be the HCM.